docs/root/configuration/http/http_filters/local_rate_limit_filter.rst

```diff
@@ -22,8 +22,9 @@ configured to be returned.
 <envoy_v3_api_field_extensions.filters.http.local_ratelimit.v3.LocalRateLimit.request_headers_to_add_when_not_enforced>` can be
 configured to be added to forwarded requests to the upstream when the local rate limit filter is enabled but not enforced.
 
-.. note::
-  The token bucket is shared across all workers, thus the rate limits are applied per Envoy process.
+Depending on the value of the config :ref:`local_rate_limit_per_downstream_connection <envoy_v3_api_field_extensions.filters.http.local_ratelimit.v3.LocalRateLimit.local_rate_limit_per_downstream_connection>`,
+the token bucket is either shared across all workers or on a per connection basis. This results in the local rate limits being applied either per Envoy process or per downstream connection.
+By default the rate limits are applied per Envoy process.
```

Comment:
It wasn't obviously clear to me why one would need this knob instead of just setting concurrent streams per connection.

Would stream limits work for your use case, or do you explicitly want to rate limit for some routes and not others?  If the latter is the case I think it'd be useful to doc up an example of how you could apply limits to routes A+B and not limit C, and how the cap would be applied.

Reply:
Mentioning this – which Alyssa pointed to offline – which might or might not be Gokul's case, but `max_concurrent_streams` is only for H2/H3.

So this would still be necessary for H1.

Reply:
But isn't this a per-connection thing?  HTTP/1 can only have one stream per connection at a time since it's just not multiplexed :-)

Reply:
I believe it comes down to the fact that the per connection quotas set and enforced via the [token bucket configuration](https://www.envoyproxy.io/docs/envoy/latest/api-v3/type/v3/token_bucket.proto#envoy-v3-api-msg-type-v3-tokenbucket) doesn't really care if the requests were concurrent. The SETTINGS_MAX_CONCURRENT_STREAMS, from what I understand, allows one to set a hard limit on the maximum number of pending or outstanding requests on the connection. On the other hand, setting a per connection rate limit quota of 25 tokens with a 5 sec fill_interval, for instance, would enforce it over that time period even if the requests were made serially one after another. If this makes sense, I can go ahead and add a blurb to this doc explaining this difference 👍